### ISSUE TITLE
After restart of browser or crash, great suspended loses preferences

### ISSUE THREAD

**Issue by rayarachelian** [deanoemcke/thegreatsuspender#468]
I have it set to suspend after 3 days.  When the browser restarts, the setting defaults back to 1hr, and nearly all of my tabs get suspended.

**Comment by deanoemcke** [comment]
Is this still happening on the latest release? (v6.30).
The upcoming release will have support for settings sync which should help.

**Comment by rayarachelian** [comment]
Hi, 
This only still happens in Chrome Incognito windows/tabs. Seems those can't store settings perhaps. 
Thanks much. 



-------- Original message --------

**Comment by utuku** [comment]
i see. Tnx

2017-07-25 22:08 GMT+03:00 Ray Arachelian <notifications@github.com>:

**Comment by deanoemcke** [comment]
Yes. It turns out that localstorage does not work in incognito mode (for obvious privacy reasons i suppose): https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API#localStorage

The new settings synching feature in the upcoming release will allow you to get around this.

**Comment by rayarachelian** [comment]
Ok makes sense. Any chance of detecting Incognito tabs and offering a checkbox to white list Incognito tabs so they don't get suspended? 

-------- Original message --------

**Comment by deanoemcke** [comment]
I'm not sure I fully understand. Incognito is a whole other window. You can not have incognito tabs mixed up with normal tabs.
Are you wanting it so that tabs in your incognito window never get suspended? This should be fairly easy. Just don't allow the extension to run in incognito mode (you can set this from the chrome://extensions url). 
Or am I missing something?

**Comment by rayarachelian** [comment]
Ah, OK. 

-------- Original message --------

**Comment by DanH42** [comment]
I think it's worth noting that this isn't an issue I encounter in most other extensions. Don't get me wrong, sync features are great to have, but enabling sync shouldn't be necessary just to persist settings in incognito windows. I use different settings on different machines, so sync doesn't make sense for me anyway.

Off the top of my head, uBlock Origin is an example of an extension that always has the same settings in and out of incognito mode. From a quick glance at the source code, it looks like it's simply using `chrome.storage.local.[get/set]`:
https://github.com/gorhill/uBlock/blob/master/platform/chromium/vapi-background.js#L91
https://github.com/gorhill/uBlock/blob/master/src/js/storage.js#L89

**Comment by deanoemcke** [comment]
@DanH42 hmm.. maybe `chrome.storage.local` is shared across normal and incognito mode.
i'm using `window.localStorage` and it does not.

I'll do some more investigation.

**Comment by JeffH1987** [comment]
In version 7.1.0 (Chrome up to date), this still seems an issue. Not only are preferences not saved in incognito mode in chrome, but also preferences to not carry over from "regular" mode.

Basically, I use Chrome about the same within and without incognito. In the normal window, I am set to ten minutes for tabs to be suspended. When I switch to incognito, settings are lost and the ones that aren't seem to be the defaults. So, tabs are suspended after only 20 sec, and whitelisting is all but meaningless from incognito session to incognito session. This forces me to change the truncated settings every time, if I remember.

As it appears other's already mentioned, most other extensions will act the same in incognito if allowed to run in such windows. Never use the Sync feature, as I am rarely signed in to my google account while browsing. In other words, this extension is borderline unusable for me when in an incognito window.

**Comment by memartinb** [comment]
Same here, v7.1.0 on Chromium 80
